File: frontend/src/api/taskTypes.ts
// L-FE-1 修复：跨目录相对路径（../../../common）脆弱，改用 @common 别名导入
import taskTypes from '@common/task-types.json'

export const TASK_TYPES = taskTypes as readonly string[]
export type TaskType = (typeof TASK_TYPES)[number]
export const TASK_TYPE_SET = new Set<string>(TASK_TYPES)

File: backend/app/constants/task_types.py
from __future__ import annotations

import json
from pathlib import Path

TASK_TYPES: tuple[str, ...] = tuple(
    json.loads(
        Path(__file__)
        .resolve()
        .parents[3]
        .joinpath("common", "task-types.json")
        .read_text(encoding="utf-8")
    )
)
TASK_TYPE_SET: frozenset[str] = frozenset(TASK_TYPES)
